uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -300,7 +343,10 @@ define([
             return loadStatuses();
           })
           .then(function () {
-            return $q.all[initUserRole(), initOpenMode()];
+            return $q.all[
+              initUserRole(),
```

Comment:
It seems that both `initUserRole` and `initOpenMode` has synchronous code inside it, but still we are using ` $q.resolve(mode)`. We can convert it to synchronous code and get rid of the promise.